src/Microsoft.Health.Fhir.Core/Features/Persistence/ResourceWrapper.cs

```diff
@@ -119,5 +119,16 @@ public void UpdateSearchIndices(IReadOnlyCollection<SearchIndexEntry> searchIndi
         {
             SearchIndices = searchIndices;
         }
+
+        public virtual bool SearchIndicesEqual(IReadOnlyCollection<SearchIndexEntry> otherIndices)
```

Comment:
nit: Do we need this as a part of ResourceWrapper class? We could just have it as a separate helper method in ReindexUtilities that does the comparison between two searchindices (since that is the only use case for now). That way we can avoid having this virtual method here.